### ISSUE TITLE
[HOLD for payment 2024-05-09] [Violations] [$500] Implement `Receipt Audit` Feature / `Note` type violations

### ISSUE THREAD

**Issue by JmillsExpensify** [Expensify/App#36288]
## Relevant Background - High-level
![image](https://github.com/Expensify/App/assets/1127863/b99879f1-23be-4448-ba87-877d20331fa5)

Notes for completeness:
 - `Note` type violations are largely “informational ” and arise from “Receipt Audit” (e.g. we scan the receipt, whether or not the employee does).
 - `Note` type violations do not generate an RBR in the LHN for admins or members, because notes are informational and there is nothing for either admin or member to fix. However, we will do the following:
   - Update the dot separator sub-state for the request preview with `Review required`
     - Note: Once the report moves to either an `approved` or `reimbursed` state, whichever comes first, the `Review required` is removed as a sub-state. This is because while the informational note remains, there is never anything in particular to resolve or dismiss, like other violations. 
 - If a receipt exists on a request and workspace where Receipt Auditing scans the receipt information in the background, Receipt Audit will appear immediately below the receipt image. 
 - Receipt Audit is shown in the “supporting text” color, along with a summary of how many (or no) issues were found
    - If one or more issues exist, an icon of a paper/magnifying glass appear to the left of the text: `Receipt Audit * %numIssues% issues found` (note: “issue” is singular if only one issue). 
    - If no issues exist, an icon of a paper overlaid with a check mark appear to the left of the text: `Receipt Audit * no issues found`

## Detailed (Note: This is outdated and proposals should note suggested updates)
We’ll update src/components/ReportActionItem/MoneyRequestView.js with a similar approach we took for showing inline violations
 - [Under](https://github.com/Expensify/App/blob/67df8e9ff334899f836080dbd7e65dc6ffc137c1/src/components/ReportActionItem/MoneyRequestView.js#L108-L116) `{hasReceipt && ( <View style={styles.moneyRequestViewImage}>`
  - We’ll add a new component ReceiptAudit and pass a prop `notes` with the value transactionVioltions.notice
  - That value will be an array with the notices for the transaction. 
  - If the array is empty, we’ll show the green we’ll show `Receipt Verified No issues found`
  - Otherwise, we’ll show 🔎 `Receipt Audit N issues found`, and the copy of the notices in red below
  - `<ReceiptAudit notice={transactionViolation.notice} />`
  - Here’s a POC version of that component
  - Note: we need to add translations for “X issues found” and “No Issues found”

```import Text from './Text';
import Icon from './Icon';
import PropTypes from 'prop-types';

const propTypes = {
    notice: PropTypes.arrayOf(PropTypes.string).required
};
const defaultProps = {
    notice: [],
};
function ReceiptAudit(notice) {
    debugger;
    return (
        <>
            <Icon width={32} height={32} src={notice.length > 0 ? Expensicons.Receipt : Expensicons.Checkmark} />
            <Text>
                Receipt Audit • {notice.length > 0 ? `${notice.length} Issue(s) Found` : 'No issues Found'}
            </Text>
        </>
    );
}```
<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~019f84ef0521d4c9c9</li>
        <li>Upwork Job ID: 1756094841883127808</li>
        <li>Last Price Increase: 2024-02-23</li>
<li>Automatic offers: </li>
<ul>
<li>Krishna2323 | Contributor | 0</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @JmillsExpensify</details>

**Comment by kevalacme** [comment]
hi

**Comment by JmillsExpensify** [comment]
@sobitneupane Thoughts on the proposals so far?

**Comment by sobitneupane** [comment]
Thanks for the proposal @neonbhai and @Krishna2323

>>Note type violations do not generate an RBR in the LHN for admins or members, because notes are informational and there is nothing for either admin or member to fix. However, we will do the following:
Update the dot separator sub-state for the request preview with Review required
Note: Once the report moves to either an approved or reimbursed state, whichever comes first, the Review required is removed as a sub-state. This is because while the informational note remains, there is never anything in particular to resolve or dismiss, like other violations.
Highlight notes in Guided Review for extra review (more in [out of scope](https://docs.google.com/document/d/1iGvpCoQ6Wt-i2wcpvaMKsWZRNs-oAsWoaS_nLue7_E0/edit#bookmark=id.homd13monm1o))


This portion of the request is missing in both the proposals.

**Comment by kevalacme** [comment]
Contributor details
Your Expensify account email: <Keval.acmeuniverse@gmail.com>
Upwork Profile Link: <https://www.upwork.com/fl/~01de57443ab6a9b794>

**Comment by JmillsExpensify** [comment]
@sobitneupane Can you help me understand what detail you're looking for? Comments on each point though. 

> `Note` type violations do not generate an RBR in the LHN for admins or members, because notes are informational and there is nothing for either admin or member to fix. 

Any questions on this. RBR is the red dot that appears next to a chat that appears in the LHN

>    - Update the dot separator sub-state for the request preview with `Review required`
>     - Note: Once the report moves to either an `approved` or `reimbursed` state, whichever comes first, the `Review required` is removed as a sub-state. This is because while the informational note remains, there is never anything in particular to resolve or dismiss, like other violations. 

Does this image help?
<img width="549" alt="Screenshot 2024-02-20 at 18 52 33" src="https://github.com/Expensify/App/assets/1127863/be88d80c-ec51-4a2a-9a28-3bb23f1eafd1">

>    - Highlight notes in Guided Review for extra review (more in [out of scope](https://docs.google.com/document/d/1iGvpCoQ6Wt-i2wcpvaMKsWZRNs-oAsWoaS_nLue7_E0/edit#bookmark=id.homd13monm1o))

This is a forward looking concern, so not relevant for this issue. I removed it from the OP above.

**Comment by sobitneupane** [comment]
Thank you for the update, @JmillsExpensify. Your provided information is sufficient for us to proceed. Much appreciated!

So, we would want to show `Review Required` in MoneyRequestPreview if there is any `Note` type violations and remove it once it is approved or reimbursed.

cc: @Krishna2323 @neonbhai

**Comment by JmillsExpensify** [comment]
Would be great if you can prioritize proposal review for this issue @sobitneupane. We're trying to wrap up our violations initiative in the coming weeks.

**Comment by sobitneupane** [comment]
Sure @JmillsExpensify. I will prioritize the issue and try to get it done asap.

**Comment by sobitneupane** [comment]
Thanks for the proposal everyone.

[Proposal](https://github.com/Expensify/App/issues/36288#issuecomment-1936980709) from @Krishna2323 looks good to me.

🎀 👀 🎀 C+ reviewed

**Comment by JmillsExpensify** [comment]
Awesome, thanks @sobitneupane! @Krishna2323 When you do think you'll be able to have a PR up for review?

**Comment by lakchote** [comment]
@Krishna2323 what's the update on this?

**Comment by lakchote** [comment]
> @JmillsExpensify, I'm on it, will surely raise PR for review tomorrow.

What's the status on this @Krishna2323? I see the PR is still in draft status.

**Comment by JmillsExpensify** [comment]
Thank you!

**Comment by sobitneupane** [comment]
@JmillsExpensify How can I produce note type violations. I think I need to be in Violations beta. Can you please add dev.sobitneupane+1@gmail.com and dev.sobitneupane+2@gmail.com in Violations beta.

**Comment by sobitneupane** [comment]
@Expensify/design team, To implement the feature, we will need the following icons.

- an icon of a paper overlaid with a check mark appear
- an icon of a paper/magnifying glass

**Comment by sobitneupane** [comment]
>  where do you see this one in the mockups?

@shawnborton It is mentioned in the issue description.

> Receipt Audit is shown in the “supporting text” color, along with a summary of how many (or no) issues were found
> - If one or more issues exist, an icon of a paper/magnifying glass appear to the left of the text: Receipt Audit * %numIssues% issues found (note: “issue” is singular if only one issue).
> - If no issues exist, **an icon of a paper overlaid with a check mark appear** to the left of the text: Receipt Audit * no issues found

**Comment by shawnborton** [comment]
Ah got it, thanks for clarifying!

**Comment by dannymcclain** [comment]
>cc @Expensify/design - I wonder if we should try to use our updated badge component for this? I think Danny had once mentioned that he wasn't a big fan of using multiple/custom badge styles but I do wonder if it makes sense to have a success/danger version of our default badge, and if we should use something like that here?

@shawnborton Yeah I do think we should at least try to adapt our new badge component to handle this. I can do some explorations for success/danger treatments (or you are more than welcome to as well!). I just don't want multiple different badges with different icon sizing and spacing and stuff. 👍 [Figgy explore](https://www.figma.com/file/GuXL2bpQRauOa5qCD3ZesG/NewDot-Violations?type=design&node-id=1133-10691&mode=design&t=LYDi7jkISVc0ELk2-11)

**Comment by JmillsExpensify** [comment]
> @JmillsExpensify How can I produce note type violations. I think I need to be in Violations beta. Can you please add [dev.sobitneupane+1@gmail.com](mailto:dev.sobitneupane+1@gmail.com) and [dev.sobitneupane+2@gmail.com](mailto:dev.sobitneupane+2@gmail.com) in Violations beta.

Done. It might take a couple of hours for the beta to propagate.

**Comment by sobitneupane** [comment]
We need Spanish Translation for following terms:

```
receiptAudit: 'Receipt Audit',
receiptVerified: 'Receipt Verified',
receiptNoIssuesFound: 'No issues found',
receiptIssuesFound: (count: number) => `${count} ${count === 1 ? 'issue' : 'issues'} found`,

```
cc: @JmillsExpensify

**Comment by JmillsExpensify** [comment]
> @JmillsExpensify Should we show those note violations only on admin side? For the same transaction, notice violations were present in admins side but not on employee/requestors side?

Good question. This one is pretty tricky, because we don't share it in certain circumstances (e.g. in Guided Review, which is a report approval feature, similar to Docusign). In any case, I'm fairly sure that any notes originating from a receipt scan are only shown to the manager/approver. @cead22 Do you mind confirming just in case I'm missing something? 

> On my testing, I also found out that we send those violations only to admin of collect workspace. So, Receipt Audit Feature will be visible only to Collect Workspace's admin, right?

Receipt Audit is only a feature for paid workspaces, so Collect and Control.

**Comment by JmillsExpensify** [comment]
> We need Spanish Translation for following terms:
> 
> ```
> receiptAudit: 'Receipt Audit',
> receiptVerified: 'Receipt Verified',
> receiptNoIssuesFound: 'No issues found',
> receiptIssuesFound: (count: number) => `${count} ${count === 1 ? 'issue' : 'issues'} found`,
> ```
> 
> cc: [@JmillsExpensify](https://github.com/JmillsExpensify)

Then for this one, there's some internal discussion around changing this language slight. Let me bump internally before I provide translations.

**Comment by lakchote** [comment]
> @JmillsExpensify, I also need access to violations beta for testing, can you pls add [belivethatkg@gmail.com](mailto:belivethatkg@gmail.com) & [dev.krishna.test@gmail.com](mailto:dev.krishna.test@gmail.com) 🙏🏻

@Krishna2323 I've added you to the violations beta with these 2 emails.

**Comment by dannymcclain** [comment]
Added mockups for the final design here: https://github.com/Expensify/App/pull/37813#issuecomment-2004153828

**Comment by cead22** [comment]
> In any case, I'm fairly sure that any notes originating from a receipt scan are only shown to the manager/approver. @cead22 Do you mind confirming just in case I'm missing something?

We show it to everyone with access to the report. Eg, I tested on dev on a report that was only shared with the submitter and manager, and it showed `Receipt not verified. Please confirm accuracy`. I then made another account a workspace admin, which automatically shares the reports on that workspace, and that account also sees the note

**Comment by sobitneupane** [comment]
https://github.com/Expensify/App/issues/36288#issuecomment-2020784641

Thanks @cead22

@JmillsExpensify So, we would want to show notes violation and review required sub-state to all parties having access to the report(requestor, approver,  admin), right? 
<img width="1440" alt="Screenshot 2024-04-01 at 17 23 12" src="https://github.com/Expensify/App/assets/25876548/9ef816a6-a311-499e-9679-0ad546a91086">
<img width="431" alt="Screenshot 2024-04-01 at 17 25 50" src="https://github.com/Expensify/App/assets/25876548/70f4fbeb-378a-4455-afc7-fbaf23a4d192">

**Comment by shawnborton** [comment]
> and review required sub-state

When the "review required" label is on the preview card, we should also show a red dot indicator on the card right?

**Comment by dannymcclain** [comment]
>When the "review required" label is on the preview card, we should also show a red dot indicator on the card right?

Yeah I think so. This should act like any other RBR violation—so the preview should also get the RBR/red dot.

**Comment by JmillsExpensify** [comment]
> @JmillsExpensify So, we would want to show notes violation and review required sub-state to all parties having access to the report(requestor, approver, admin), right?

Yes

**Comment by sobitneupane** [comment]
> 
> When the "review required" label is on the preview card, we should also show a red dot indicator on the card right?

The issue description says we don't want to generate RBR, that's the reason for not showing ret dot indicator.

> Note type violations do not generate an RBR in the LHN for admins or members, because notes are informational and there is nothing for either admin or member to fix. However, we will do the following:

**Comment by JmillsExpensify** [comment]
@sobitneupane RBR and the red dot are different. For instance, if someone approves a report then the violation doesn't go away, the red dot will always appear because a violation/rule has been broken.

**Comment by JmillsExpensify** [comment]
Basically this is the ![CleanShot 2024-04-05 at 11 04 47@2x](https://github.com/Expensify/App/assets/1127863/20fb3586-8aa3-45a2-8822-b92fa1024ddb). Let me know if you have any questions about how it works.

**Comment by sobitneupane** [comment]
Thanks, @JmillsExpensify!

Should red dot appear only on the preview card, or should it also be visible in the transaction thread on the LHN? If we're considering implementing it in the LHN as well, I believe it would essentially be an RBR. Please correct me if I'm mistaken.

**Comment by JmillsExpensify** [comment]
I believe that is answered in the screenshot above. No red dot visible in the LHN, so no RBR.

**Comment by JmillsExpensify** [comment]
For the sake of clarity though, the only time RBR should appear for the transaction thread is when both and these statements are true:

 - There are violations AND
 - Submitter can still make edits (either `open` or `processing` report, depending on the configuration of the workspace. 

If both of those statements are not true, then RBR won't show in the LHN for the submitter. Further the transaction thread is never RBR for the manager or the admin/owner.

**Comment by JmillsExpensify** [comment]
Payment summary:

 - Contributor: @Krishna2323 $500 
 - Contributor+: @sobitneupane $500

**Comment by JmillsExpensify** [comment]
Contributor contract paid out via Upwork. C+ is paid via New Expensify, so I'm going to go ahead and close the issue and circle back when I receive a payment request. @sobitneupane Don't forget to get the BZ checklist completed. I definitely think we need a series of regression tests for this feature.

**Comment by sobitneupane** [comment]
Oops I missed the issue.
**Regression Test Proposal:**

1. Submit an expense to a paid policy group with a valid receipt.
2. Wait for the receipt to be scanned and then verify that the "Receipt Verified" message is shown above the receipt.
3. Change the amount to something higher than the scanned receipt.
4. Verify that: 
         - The "Issue Found" message is displayed above the receipt.
         - A violation message is displayed below the Amount field.
         - The "Review required" substate is displayed in the Request Preview.
5. Change the date field.
6. Verify that:
         - The "Issue Found" header is displayed above the receipt.
         - A violation message is displayed below the Date field.
         - The "Review required" substate is displayed in the Request Preview.
7.  If the receipt scan is not successful, the "Receipt not verified. Please confirm accuracy." violation message should be shown below the receipt with the "Issue Found" message above the receipt and "Review required" substate in Request Preview.

Do we agree 👍 or 👎

**Comment by cead22** [comment]
@JmillsExpensify I can add a test to the violations test script based on this ^

**Comment by JmillsExpensify** [comment]
Ok thanks @cead22! Sounds good. 

I need to have someone else confirm this payment summary that isn't me, so I reached out to someone else on the BZ team.

**Comment by mallenexpensify** [comment]
Contributor: @Krishna2323 paid $500 via Upwork 
Contributor+: @sobitneupane owed $500 via NewDot

**Comment by JmillsExpensify** [comment]
$500 approved for @sobitneupane
